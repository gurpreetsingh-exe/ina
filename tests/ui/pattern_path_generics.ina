// COMPILE_FAIL

using Option::*;
type Option[T] = Some(T) | None

fn main() {
    match Some("foo") {
        None[i32] => {},
        Some(a) => {},
    }
}
