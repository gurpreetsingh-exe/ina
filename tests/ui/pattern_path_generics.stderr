error: mismatch types
   ╭─[tests/ui/pattern_path_generics.ina:8:9]
   │
 8 │         None[i32] => {},
   │         ────┬────
   │             ╰───── expected `str`, found `i32`
   ∙
───╯
